### ISSUE TITLE
Query with .Contains only works for array, should include List<T>.

### ISSUE THREAD

**Issue by VagyokC4** [redis/redis-om-dotnet#258]
@slorello89 , Can we expand the new Contains functionality to also work with a List<T> and not just array[]?

This doesn't work
```csharp
  var hayStack = new List<string>(){ "test", "test2", "test3" };
  var needles = await needlesCollection.Where(x=> hayStack.Contains(x.Name)).ToListAsync();
```
but this does
```csharp
  var hayStack = new List<string>(){ "test", "test2", "test3" };
  var hayStackArray = hayStack.ToArray();
  var needles = await needlesCollection.Where(x=> hayStackArray.Contains(x.Name)).ToListAsync();
```

**Comment by iozcelik** [comment]
I think it is related with #203

**Comment by slorello89** [comment]
Closed by #258 - as it turns out a list would bind to the `Object` property of the Method Expression, whereas an array, being a primitive type, would not. reordering how we were interrogating the Contains method expression fixed the issue - I think the unit tests in #258 are more or less what you were trying to do here.
